### ISSUE TITLE
--buildArg flag doesn't work for docker-compose.yml projects

### ISSUE THREAD

**Issue by shaunmulligan** [balena-io/balena-cli#1053]
- **Cli version:** v9.6.0
- **Standalone install:** no
- **Node version:** v10.13.0
- **Npm version:** v6.4.1
- **Operating system and architecture:** macOS 10.13.6 

For both `build` and `deploy` the `--buildArg` flag which allows you to define `ARG` variables in your dockerfile only applies when you have a single container application. If one specifies a docker-compose.yml the --buildArg never propagates to any of the services. Below is a simple project to reproduce the issue. 
[testBuildArgs.zip](https://github.com/balena-io/balena-cli/files/2709118/testBuildArgs.zip)

If one runs `balena deploy nuc-staging --nocache --build --buildArg ACCESS_KEY=mySecret --logs` from the root of `testBuildArgs` folder, the `mySecret` variable will not be echo'd in the build logs, however, if the same command is run from `testBuildArgs/module1` folder the the `mySecret` is echo'd as expected.

<img src="https://frontapp.com/assets/img/icons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_g9bv)

**Comment by pdcastro** [comment]
@Wachiwi, I haven't started working on this issue yet, so I'd be very happy for you to fix it! I've myself only recently started getting familiar with the relevant codebase. The hints I can think of at the moment are simply links to some lines of code where the build args are manipulated.

The `balena build` and `balena deploy` commands will call `compose.buildProject` at the following lines of code:

* https://github.com/balena-io/balena-cli/blob/master/lib/actions/build.coffee#L30
* https://github.com/balena-io/balena-cli/blob/master/lib/actions/deploy.coffee#L49

Those calls land here:

* https://github.com/balena-io/balena-cli/blob/master/lib/utils/compose.coffee#L162

And eventually reach:

* https://github.com/balena-io/balena-cli/blob/master/lib/utils/compose.coffee#L222
```js
	if d.image.context?.args?
		task.dockerOpts.buildargs ?= {}
		_.merge(task.dockerOpts.buildargs, d.image.context.args)
```

And eventually call `builder.performBuilds`:

* https://github.com/balena-io/balena-cli/blob/master/lib/utils/compose.coffee#L270

Now, that call lands in an imported repo/package:

* https://www.npmjs.com/package/resin-multibuild
* https://github.com/balena-io-modules/resin-multibuild/blob/master/lib/index.ts#L157

And eventually will reach:

* https://github.com/balena-io-modules/resin-multibuild/blob/master/lib/build.ts#L98
* https://github.com/balena-io-modules/resin-multibuild/blob/master/lib/build.ts#L60

My gut feeling is that the issue and fix will actually fall on `resin-multibuild` rather than `balena-cli`.

By the way, there is work going on in `resin-multibuild` at the moment that may or may not affect this issue:

* https://github.com/balena-io-modules/resin-multibuild/pull/38/files

**Comment by pdcastro** [comment]
This issue was fixed in CLI v12.25.3. It turned out that no changes were needed to `resin-multibuild` -- the bug was in `balena-cli` itself (commit https://github.com/balena-io/balena-cli/pull/2069/commits/3199f15662373ca53fe1c7541259b31799e42315).
